File: Code-CSharp/Base/Containers/Stack/StackEntry.cs
namespace CodeCsharp.Base.Containers.Stack;

public class StackEntry {
  public static void StackProgram() {
    //initialize a stack
    Stack<string> stack = new Stack<string>();
    //initialize a stack with value
    Stack<string> stack1 = new Stack<string>(new[] { "aa", "bb", "cc", "dd" });
    //copy stack
    stack = new Stack<string>(stack1.Reverse());
    //whether stack is empty
    bool isEmpty = stack.Count == 0;
    //clear stack
    stack.Clear();
    //insert element at first
    stack.Push("ee");
    //remove first element
    stack.Pop();
    //get size of stack
    int size = stack.Count;
    //whether two stacks equal
    bool b1 = stack.Equals(stack1);
    //get first element
    string element = stack.Peek();
  }
  
  public static bool TestAll() {
    bool ret = true;
    Stack<int> c, c1, c2, c3, c4;
    c1 = new Stack<int>();
    c = new Stack<int>(new List<int> {2, 4, 3, 1, 5});
    c2 = new Stack<int>(c.Reverse());
    c.Push(6);
    c3 = new Stack<int>(c.Reverse());
    c.Pop();
    c4 = new Stack<int>(c.Reverse());
    ret = c1.Count == 0
          && c4.Count == 5
          && c.Peek() == 5
          && c2.SequenceEqual(
              new Stack<int>(new List<int> { 2, 4, 3, 1, 5 }))
          && c3.SequenceEqual(
              new Stack<int>(new List<int> {2, 4, 3, 1, 5, 6 }))
          && c4.SequenceEqual(
              new Stack<int>(new List<int>  {2, 4, 3, 1, 5 }));
    if (!ret)
      Console.WriteLine("Stack Test Failed!");
    return ret;
  }
}

File: Code-Java/src/codejava/base/containers/stack/StackEntry.java
package codejava.base.containers.stack;

import java.util.ArrayDeque;
import java.util.Arrays;

public class StackEntry {
  public static void StackProgram() {
    //initialize a stack
    ArrayDeque<String> stack = new ArrayDeque<>();
    //initialize a stack with value
    ArrayDeque<String> stack1 = new ArrayDeque<>(Arrays.asList("aa", "bb", "cc", "dd"));
    //copy stack
    stack = new ArrayDeque<>(stack1);
    //whether stack is empty
    boolean isEmpty = stack.isEmpty();
    //clear stack
    stack.clear();
    //insert element at first
    stack.push("ee");
    //remove first element
    stack.pop();
    //get size of stack
    int size = stack.size();
    //whether two stacks equal
    boolean b1 = Arrays.equals(stack.toArray(), stack1.toArray());
    //get first element
    String element = stack.peek();
  }

  public static boolean testAll() {
    boolean ret = true;
    ArrayDeque<Integer> c, c1, c2, c3, c4;
    c1 = new ArrayDeque<>();
    c = new ArrayDeque<Integer>(Arrays.asList(5, 1, 3, 4, 2));
    c2 = new ArrayDeque<>(c);
    c.push(6);
    c3 = new ArrayDeque<>(c);
    c.pop();
    c4 = new ArrayDeque<>(c);
    ret = c1.isEmpty()
        && c4.size() == 5
        && c.peek() == 5
        && Arrays.equals(c2.toArray(),
        new ArrayDeque<Integer>(Arrays.asList(5, 1, 3, 4, 2)).toArray())
        && Arrays.equals(c3.toArray(),
        new ArrayDeque<Integer>(Arrays.asList(6, 5, 1, 3, 4, 2)).toArray())
        && Arrays.equals(c4.toArray(),
        new ArrayDeque<Integer>(Arrays.asList(5, 1, 3, 4, 2)).toArray());
    if (!ret)
      System.out.println("Stack Test Failed!");
    return ret;
  }
}
